router/core/websocket_test.go

```diff
@@ -0,0 +1,186 @@
+package core
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"net"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wundergraph/cosmo/router/internal/wsproto"
+	"github.com/wundergraph/cosmo/router/pkg/config"
+	"go.uber.org/zap"
+)
+
+type MockConn struct {
```

Comment:
Typically, I would utilize tools like gomock, testify/mock, or mockery to automatically generate mocks for improved testing. However, since this project does not currently use these tools, I chose to manually create the mocks to avoid introducing additional external dependencies.